### PR TITLE
add option to ignore stacktraces

### PR DESCRIPTION
**Pull request by KristofferC**
This makes it feasible to run doctests as a part of CI on e.g. Base Julai

### CODE REVIEW

**Comment by KristofferC** [comment]
I dont think failures on maser is because of this PR

File: src/Documenter.jl [hunk 1]
```diff
@@ -174,6 +174,9 @@ ignored.
 **`strict`** -- [`makedocs`](@ref) fails the build right before rendering if it encountered
 any errors with the document in the previous build phases.
 
+**`doctest_stacktraces`** -- [`makedocs`](@ref) do not consider an erronous stacktrace
```
Comment (mortenpi) [inline src/Documenter.jl]: Typo: "erroneous"

**Comment by KristofferC** [comment]
Having an extensible way of adding options to doctests-blocks seems like a good idea to me. Something like `jldoctest key1, key2=val2, key3` etc, where `val`s are potential settings associated with the key. This would of course only be worth it if we expect to add more of these options for doctests.

**Comment by fredrikekre** [comment]
Named doctests could be changed to use `name = nameofblock` so there is one more option together with `stacktraces = {ignore, warn, error}`.

**Comment by KristofferC** [comment]
Perhaps globaal doctest settings could be specified in `DocTestSetup`?

**Comment by mortenpi** [comment]
Do you think we could have this as a regex pattern with the option from #549?

**Comment by KristofferC** [comment]
I've been meaning to write a more extensible solution than this, where you can easily specify doctest-option on a doctest by doctest basis, or globally etc.If that is something that sounds good, I will probably have time for it in ~2 weeks.

**Comment by mortenpi** [comment]
The per-doctest thing sounds great. I do think that it should ideally be specified at the doctest itself. But a global option is an OK stopgap as well I think.

But that's actually orthogonal to the question whether we should have a more generic API for dealing with changing doctests. We could

* have an option specific to stacktraces (as is here), and separate ones for other ones
* use regex patterns (i.e. #549)
* allow the user define arbitrary functions (that would remove parts that should not be checked from the expected and real doctest outputs?)

With the more generic approach, we could maintain a few more commonly used patterns/functions (e.g. one for stacktraces) in Documenter.

I am always leaning towards the most generic solution, but I am wondering whether I am over-engineering this.

cc @bicycle1885, because of your work on #549

**Comment by KristofferC** [comment]
https://github.com/JuliaDocs/Documenter.jl/pull/573

**Pull request closed without merging**